packages/ipfs-message-port-server/src/server.js

```diff
@@ -212,11 +213,14 @@ exports.Server = class Server {
     if (!query.signal.aborted) {
       try {
         const value = await query.result
-        const transfer = [...new Set(value.transfer || [])]
+        const transfer = ensureNonDetachedUniqueBuffers(value.transfer || [])
+
+        // Don't need the transfer value in the result
         delete value.transfer
 
         port.postMessage(
           { type: 'result', id, result: { ok: true, value } },
+          // @ts-ignore - Typescript expects the transfer list to be an Array, but it can actually be any iterable.
```

Comment:
I think it's better to use `@ts-expect-error` so that when ts fixes it we can remove those comments.

```suggestion
          // @ts-expect-error - Typescript expects the transfer list to be an Array, but it can actually be any iterable.
```